src/main/java/net/neoforged/neoforge/attachment/AttachmentType.java

```diff
@@ -56,43 +52,41 @@
  * </ul>
  */
 public final class AttachmentType<T> {
+    public final Class<T> dataType;
```

Comment:
What is this `dataType` actually used for? You're using a lot of typetools wackiness -- which is _notably_ fragile, and prevents people from doing stuff with generics in ways that typetools can't detect -- to get it but it doesn't seem to actually be _used_ anywhere unless I've missed something?